main/src/main/java/cgeo/geocaching/utils/FilterUtils.java

```diff
@@ -40,6 +41,21 @@ public static void openFilterActivity(final Activity activity, final GeocacheFil
                 filteredList, true);
     }
 
+    @Nullable
+    private static Pair<Pair<TextParam, TextParam>, GeocacheFilter> getNeutralActionForContext(final GeocacheFilterContext filterContext) {
```

Comment:
To transfer three parameters use class  `ImmutableTriple` instead of nested `Pairs`

Context-wise: if I understand code correctly, then "last/previous filter" choice is only given when no filter is currently selected. Is this correct? Why not offering this option always?

Reply:
> To transfer three parameters use class  `ImmutableTriple` instead of nested `Pairs`


Good point, didn't had that in mind


> 
> Context-wise: if I understand code correctly, then "last/previous filter" choice is only given when no filter is currently selected. Is this correct? Why not offering this option always?
> 
> 
See comment 
https://github.com/cgeo/cgeo/pull/16524#issuecomment-2571794976

Reply:
Used `ImmutableTriple`

Reply:
I will have a look into showing it always

Reply:
To get it work, I had to avoid that the negative button is removed automatically...